checks/evaluation/branch_protection.go

```diff
@@ -103,77 +103,37 @@ func BranchProtection(name string, dl checker.DetailLogger,
 	}
 }
 
-func computeNonAdminBasicScore(scores []levelScore) int {
-	score := 0
-	for i := range scores {
-		s := scores[i]
-		score += s.scores.basic
-	}
-	return score
-}
-
-func computeNonAdminReviewScore(scores []levelScore) int {
-	score := 0
-	for i := range scores {
-		s := scores[i]
-		score += s.scores.review
-	}
-	return score
-}
-
-func computeAdminReviewScore(scores []levelScore) int {
-	score := 0
-	for i := range scores {
-		s := scores[i]
-		score += s.scores.adminReview
-	}
-	return score
-}
-
-func computeNonAdminThoroughReviewScore(scores []levelScore) int {
-	score := 0
-	for i := range scores {
-		s := scores[i]
-		score += s.scores.thoroughReview
-	}
-	return score
-}
-
-func computeAdminThoroughReviewScore(scores []levelScore) int {
-	score := 0
-	for i := range scores {
-		s := scores[i]
-		score += s.scores.adminThoroughReview
-	}
-	return score
-}
-
-func computeNonAdminContextScore(scores []levelScore) int {
-	score := 0
-	for i := range scores {
-		s := scores[i]
-		score += s.scores.context
-	}
-	return score
-}
-
-func computeCodeownerThoroughReviewScore(scores []levelScore) int {
-	score := 0
-	for i := range scores {
-		s := scores[i]
-		score += s.scores.codeownerReview
+func sumUpScoreForTier(tier int, scoresData []levelScore, dl checker.DetailLogger) int {
```

Comment:
Or might it not be better to keep the old functions, one for each tier? With this PR, we have special-case logic for each tier in this function AND "repetitive" special-case logic for each tier in `computeFinalScore`. Previously, the logic for each tier was stored in a dedicated function (with "repetitive" special-case logic in `computeScore`).

Having this single function reduces the code duplication in all those specialized functions, but I think the legibility is a lower. (lambda functions would be useful here, but they aren't feasible/idiomatic in Go...)

I do think a single loop for Tiers 2 and 4 is better than the double loop they currently use (one for each score), though... Though maybe that's intentional, thinking about probes and policies?

Reply:
I liked the idea of using the Enums! And it should also eliminate the need to treat the case of having a invalid input (as you pointed in https://github.com/ossf/scorecard/pull/3499#discussion_r1414262038)

Now concerning the need choice of keeping the old function, I like the idea of having the function `sumUpScoresForTier` because, despite carrying special-case logic for each tier, it has a very clear and simple purpose. 

My motivations for this change were the duplications, of course, but also the fact that I was getting confused with the calls of those functions with very similar namings. Another problem was that when I read something like `computeNonAdminBasicScore`, I usually though it would do something way more complex than just summing up the scores.

So I'd say it's fine to keep the different functions and the duplicated code if you prefer, but I think at least the naming should be reviewed.

Reply:
@spencerschrock can you share your thoughts before I work any further on this?

Reply:
with the pending Branch Protection probe conversion, this evaluation code is likely to change. I wouldn't spend time perfecting it in this PR. So either revert this tier scoring portion of this change, or leave it as is.